### PR TITLE
chain: update all error messages

### LINKED ISSUE #758: Add `CodeItemNotFound` error and patch error messages
This is whenever people specify unknown things like request id, oracle script id, etc.

### LINKED ISSUE #759: Add 'CodeItemDuplication` error and patch error messages
Whenever someone trying to do something more than once. Like reporting.

### LINKED ISSUE #760: Add 'CodeUnauthorizedPermission` error and patch error messages
When you are the owner, but you try to do something

### LINKED ISSUE #761: Make sure we have good error messages in all places
This is the finale ticket. Consult Swit for error code.

### PR DESCRIPTION
**Pull request by sorawit**
Fixes #758 
Fixes #759 
Fixes #760 
Fixes #761

### CODE REVIEW

File: chain/x/zoracle/handler.go [hunk 1]
```diff
@@ -106,21 +106,20 @@ func addUint64Overflow(a, b uint64) (uint64, bool) {
...
 	endBlockExecuteGasLimit := keeper.EndBlockExecuteGasLimit(ctx)
 	gasConsumed := uint64(0)
-
 	firstUnresolvedRequestIndex := len(pendingList)
 
 	for i, requestID := range pendingList {
 		request, err := keeper.GetRequest(ctx, requestID)
-		if err != nil {
-			// Don't expect to happen
+		if err != nil { // should never happen
```
Comment (sorawit) [inline chain/x/zoracle/handler.go:115]: @taobun Please help check this. This is the only logic change.

File: chain/x/zoracle/internal/types/result.go [hunk 2]
```diff
@@ -35,9 +35,9 @@ func NewResult(
 }
 
 // DecodeResult is a helper function for decoding bytes to Result.
-func DecodeResult(b []byte) (Result, error) {
+func MustDecodeResult(b []byte) Result {
 	if len(b) < 40 {
-		return Result{}, fmt.Errorf(fmt.Sprintf("Expect size of input to be at least 40 bytes but got %d bytes", len(b)))
+		panic(fmt.Errorf("Expect size of input to be at least 40 bytes but got %d bytes", len(b)))
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/result.go:41]: Also here. Don't see why we should return error.

File: chain/x/zoracle/internal/types/error.go [hunk 3]
```diff
@@ -14,69 +14,36 @@ const (
...
-	CodeRequestNotFound    sdk.CodeType = 103
-	CodeResultNotFound     sdk.CodeType = 104
-	CodeInvalidOwner       sdk.CodeType = 105
-	CodeOutOfReportPeriod  sdk.CodeType = 106
-	CodeDuplicateValidator sdk.CodeType = 107
-	CodeDuplicateRequest   sdk.CodeType = 108
-	CodeReportNotFound     sdk.CodeType = 109
-
-	WasmError sdk.CodeType = 105
+	CodeItemDuplication        sdk.CodeType = 204
```
Comment (taobun) [inline chain/x/zoracle/internal/types/error.go]: After removing all old errors, we can change new error to 10X?

File: chain/x/zoracle/handler.go (hunk 1 above)
Comment (taobun) [reply chain/x/zoracle/handler.go:115]: Make sense to set fail.

File: chain/x/zoracle/internal/types/result.go [hunk 4]
```diff
@@ -35,9 +35,9 @@ func NewResult(
 }
 
 // DecodeResult is a helper function for decoding bytes to Result.
-func DecodeResult(b []byte) (Result, error) {
+func MustDecodeResult(b []byte) Result {
```
Comment (taobun) [inline chain/x/zoracle/internal/types/result.go:39]: Update comment

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/internal/types/result.go [hunk 5]
```diff
@@ -34,10 +34,11 @@ func NewResult(
 	}
 }
 
-// DecodeResult is a helper function for decoding bytes to Result.
-func DecodeResult(b []byte) (Result, error) {
+// DecodeResult is a helper function for decoding bytes to Result. The function panics if the given
```
Comment (taobun) [inline chain/x/zoracle/internal/types/result.go]: Must!!!

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/internal/types/result.go (hunk 5 above)
Comment (sorawit) [reply chain/x/zoracle/internal/types/result.go]: Right. I misunderstood your comment earliler. Thanks for reminding again.

**Review by taobun — APPROVED** [review]

**Pull request merged**